scripts/cluster/join.py

```diff
@@ -730,6 +752,10 @@ def join_dqlite(connection_parts):
     master_ep = connection_parts[0].split(":")
     master_ip = master_ep[0]
     master_port = master_ep[1]
+
+    if len(connection_parts) > 2 and verify:
```

Comment:
@balchua if you specify a token that does not include a '/' it would be as if there is no sha to verify the connection and the `verify_server` will not be called.

Reply:
Im afraid the message printed out during `add-node` maybe wrong if token is passed.  I think a check is needed prior to executing `openssl` and verify if `token` is passed in as argument.

Reply:
@balchua the message produced if a user provided token is requested looks like this:
```
root@u1:~# microk8s.add-node --token 12345678901234567890123456789012
From the node you wish to join to this cluster, run the following:
microk8s join 10.228.72.149:25000/12345678901234567890123456789012/d8d3e9e758d011b0b180122d55173cb3ebf426d4332cb9bea5ba0e464560280e

If the node you are adding is not reachable through the default interface you can use one of the following:
 microk8s join 10.1.215.192:25000/12345678901234567890123456789012/d8d3e9e758d011b0b180122d55173cb3ebf426d4332cb9bea5ba0e464560280e
 microk8s join 10.1.100.0:25000/12345678901234567890123456789012/d8d3e9e758d011b0b180122d55173cb3ebf426d4332cb9bea5ba0e464560280e
 microk8s join 10.228.72.149:25000/12345678901234567890123456789012/d8d3e9e758d011b0b180122d55173cb3ebf426d4332cb9bea5ba0e464560280e
```

This looks correct to me as next to the token we add the certificate token. However, you are right the connection string is not only the token provided. We will be able to still join if we use the connection string without the server certificate fingerprint, ie:
```
microk8s join 10.228.72.149:25000/12345678901234567890123456789012
```
Alternatively there is the `--skip-verify` flag
```
microk8s join 10.228.72.149:25000/12345678901234567890123456789012/d8d3e9e758d011b0b180122d55173cb3ebf426d4332cb9bea5ba0e464560280e --skip-verify
```

Would this be ok with you?